configuration/configmgr/configmgrlib/mod_template_support/Variables.cpp

```diff
@@ -90,16 +90,20 @@ std::string Variables::doValueSubstitution(const std::string &value) const
     std::size_t bracesStartPos = result.find("{{");
     bool done = bracesStartPos == std::string::npos;
 
+    // {{hwips[{{numsystems}}-{{ccnum}}]}}
+    // ^                                ^
+    // 0                                33
+
     while (!done)
     {
         index = m_curIndex;
         std::size_t bracesEndPos = findClosingDelimiter(result, bracesStartPos,"{{", "}}");
-        varName = result.substr(bracesStartPos + 2, bracesEndPos - bracesStartPos - 2);
+        varName = result.substr(bracesStartPos + 2, bracesEndPos - bracesStartPos - 2);  // hwips[{{numsystems}}-{{ccnum}}]
 
         //
         // If there is an index defined, evaluate it and update the index to be used for the final value
-        std::size_t bracketStartPos = result.find('[');
-        std::size_t sizePos = result.find(".size");
+        std::size_t bracketStartPos = varName.find('[');  // 5
+        std::size_t sizePos = varName.find(".size");  // npos
```

Comment:
trivial: not sure those comments are helpful/meaningful to other reads

Reply:
Again, left over debugging comments. Removed along with some others I found.